src/sandstorm/activity.capnp

```diff
@@ -0,0 +1,180 @@
+# Sandstorm - Personal Cloud Sandbox
+# Copyright (c) 2016 Sandstorm Development Group, Inc. and contributors
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+@0xa4e001d4cbcf33fa;
+
+$import "/capnp/c++.capnp".namespace("sandstorm");
+
+using Util = import "util.capnp";
+
+struct ActivityEvent {
+  # Describes an event in a grain's activity feed / log.
+  #
+  # Call SessionContext.activity() to post new activity. The activity is attributed to the user
+  # whose SessionContext was used.
+
+  path @0 :Text;
+  # Path (as in, URL) within the grain where the user can see this activity. Users inspecting the
+  # event will be linked to this location. Visiting this path implicitly marks the event as "read".
+  #
+  # This should NOT include a leading '/'. (So, an empty string -- the default -- goes to the grain
+  # root).
+  #
+  # Users are able to subscribe to events at specific paths to get notified on activity.
```

Comment:
Revised in commit: 34ca052

Now there is an explicit notion of threads and of subscriptions. This seems to improve the Etherpad use case considerably: previously I implemented it such that a comment author was implicitly "mentioned" by any reply, but that wasn't really right: "mentions" should actually be the people who are literally @-mentioned or tagged in some way. The new threads/subscriptions semantics means I don't have to overload mentions now. (Though as of this writing I still need to actually implement the interface.)